lib/rubocop/cop/style/commented_keyword.rb

```diff
@@ -92,8 +92,22 @@ def source_line(comment)
           comment.source_range.source_line
         end
 
+        def definition_type(line)
+          return :subclass if line.start_with?(/\A\s*class\s+\w+\s*<\s*\w+/)
+          return :method if line.start_with?(/\A\s*def\s/)
+
+          :others
+        end
+
         def rbs_inline_annotation?(line, comment)
-          comment.text.start_with?('#:') && line.start_with?(/\A\s*def\s/)
+          case definition_type(line)
+          when :subclass
+            comment.text.start_with?(/#\[.+\]/)
+          when :method
+            comment.text.start_with?('#:')
+          else
+            false
+          end
```

Comment:
Can this be refactored as follows?

```diff
diff --git a/lib/rubocop/cop/style/commented_keyword.rb b/lib/rubocop/cop/style/commented_keyword.rb
index eae7e920a..17ec063d7 100644
--- a/lib/rubocop/cop/style/commented_keyword.rb
+++ b/lib/rubocop/cop/style/commented_keyword.rb
@@ -57,6 +57,9 @@ module RuboCop

         REGEXP = /(?<keyword>\S+).*#/.freeze

+        SUBCLASS_DEFINITION = /\A\s*class\s+\w+\s*<\s*\w+/.freeze
+        METHOD_DEFINITION = /\A\s*def\s/.freeze
+
         def on_new_investigation
           processed_source.comments.each do |comment|
             next unless offensive?(comment) && (match = source_line(comment).match(REGEXP))
@@ -92,18 +95,11 @@ module RuboCop
           comment.source_range.source_line
         end

-        def definition_type(line)
-          return :subclass if line.start_with?(/\A\s*class\s+\w+\s*<\s*\w+/)
-          return :method if line.start_with?(/\A\s*def\s/)
-
-          :others
-        end
-
         def rbs_inline_annotation?(line, comment)
-          case definition_type(line)
-          when :subclass
+          case line
+          when SUBCLASS_DEFINITION
             comment.text.start_with?(/#\[.+\]/)
-          when :method
+          when METHOD_DEFINITION
             comment.text.start_with?('#:')
           else
             false
```

Reply:
@koic 

Thanks for the suggestion. I adopted it because it writes so beautifully.